src/line_printer.cc

```diff
@@ -78,6 +84,14 @@ void LinePrinter::Print(string to_print, LineType type) {
 
   if (smart_terminal_ && type == ELIDE) {
 #ifdef _WIN32
+    if (console_ == INVALID_HANDLE_VALUE) {
+      // Assume we are in a terminal emulator.
+      to_print = ElideMiddle(to_print, 80);
```

Comment:
80 is wrong though.

Reply:
There is no way of knowing the number of columns of the terminal emulator, 80 looks like a reasonable default to me.